src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -327,28 +330,20 @@ ucs_status_t uct_tcp_sockcm_ep_server_handle_data_received(uct_tcp_sockcm_ep_t *
 {
     uct_tcp_sockcm_priv_data_hdr_t *hdr = (uct_tcp_sockcm_priv_data_hdr_t *)
                                            cep->comm_ctx.buf;
+    uint16_t client_state;
     ucs_status_t status;
     int events;
 
-    if (cep->state == (UCT_TCP_SOCKCM_EP_ON_SERVER |
-                       UCT_TCP_SOCKCM_EP_DATA_RECEIVED)) {
-        status = uct_tcp_sockcm_ep_server_invoke_conn_req_cb(cep);
-        if (status != UCS_OK) {
-            goto out;
-        }
-
-        /* server to send its private data to the client and listen to disconnect events */
-        events = UCS_EVENT_SET_EVWRITE | UCS_EVENT_SET_EVREAD;
-    } else if (cep->state == (UCT_TCP_SOCKCM_EP_SERVER_CONN_CREATED |
-                              UCT_TCP_SOCKCM_EP_DATA_RECEIVED       |
-                              UCT_TCP_SOCKCM_EP_CONN_SENT)) {
+    if (ucs_test_all_flags(cep->state, UCT_TCP_SOCKCM_EP_SERVER_CONN_CREATED |
+                                       UCT_TCP_SOCKCM_EP_DATA_RECEIVED       |
+                                       UCT_TCP_SOCKCM_EP_CONN_SENT)) {
         /* the server verifies the client's state */
-        if (*(uint16_t*)(hdr + 1) != (UCT_TCP_SOCKCM_EP_ON_CLIENT     |
-                                      UCT_TCP_SOCKCM_EP_CONN_SENT     |
-                                      UCT_TCP_SOCKCM_EP_DATA_RECEIVED |
-                                      UCT_TCP_SOCKCM_EP_CONN_NOTIFY_CB_INVOKED)) {
-            ucs_error("server received wrong client state = %d",
-                      *(uint16_t*)(hdr + 1));
+        client_state = *(uint16_t*)(hdr + 1);
```

Comment:
1. add comment that we receive "notify" here
2. can we use uct_tcp_sockcm_priv_data_hdr_t here?

Reply:
2. what do you mean? this is the type of hdr.

Reply:
instead of directly using uint16_t, define a header type for "notify" message